### PR TITLE
test: MET-2190 add automation test for block

### PR DESCRIPTION
**Pull request by Sotatek-TaiTruong**
## Description

 Add automation test for block

### Issue ticket number and link

- [x] This PR has a valid ticket number or issue: [[link]](https://cardanofoundation.atlassian.net/browse/MET-2190)

### Testing & Validation

- [ ] This PR has been tested/validated in Chrome, Firefox, Safari and Brave browsers.
- [ ] The code has been tested locally with test coverage match expectations.
- [ ] Added new Unit/Component testing (if relevant).

### Security

- [ ] No secrets are being committed (i.e. credentials, PII)
- [ ] This PR does not have any significant security implications

### Code Review

- [ ] There is no unused functionality or blocks of commented out code (otherwise, please explain below)
- [ ] In addition to this PR, all relevant documentation (e.g. Confluence / README.md file) and architecture diagrams (e.g. Miro) were updated

### Design Review

- [ ] If this PR contains changes to the UI, it has gone through a design review with UX Designer or Product owner.
- [ ] In case PR contains changes to the UI, add some screenshots to notice the differences
---
##### _Before_

image

##### _After_

[comment]: <> (Add screenshots)

##### _Before_

[comment]: <> (Add screenshots)

##### _After_

[comment]: <> (Add screenshots)

##### _Before_

[comment]: <> (Add screenshots)

##### _After_

[comment]: <> (Add screenshots)

### CODE REVIEW

File: playwright/pages/blocks-dashboard.page.ts [hunk 1]
```diff
@@ -1,15 +1,252 @@
...
+  const blocksTableValueEpoch = page.getByTestId("blocks.table.value.epoch#2");
+  const blocksTableTitleSlot = page.getByTestId("blocks.table.title.slot");
+  const blocksTableTitleAbsSlot = page.getByTestId("blocks.table.title.absoluteSlot");
+  const blocksTableTitleCreateAt = page.getByTestId("blocks.table.title.createAt");
+  const blocksTableTitleTransactionsCount = page.getByTestId("blocks.table.title.transactions");
+  const blocksTableTitleFee = page.getByTestId("blocks.table.title.fee");
+  const blocksTableTitleOutput = page.getByTestId("blocks.table.title.output");
+  const blocksTableValueSlot = page.getByTestId("blocks.table.value.slot#2");
+
+  //Block detail
```
Comment (Luisdzc) [inline playwright/pages/blocks-dashboard.page.ts:31]: As I mentioned in the epochs test this block detail page locators and respective functions that are executed over the block detail page should be in the block-detail.page.ts class

File: playwright/pages/blocks-dashboard.page.ts [hunk 2]
```diff
@@ -1,15 +1,252 @@
...
+  const blocksDetailHeader = page.getByTestId("block.detail.header");
+  const blocksDetailProducer = page.getByTestId("block.detail.overview.value.producer");
+  const blocksDetailCreateTime = page.getByTestId("block.detail.overview.value.createAt");
+  const blocksDetailConfirmation = page.getByTestId("block.detail.overview.value.confirmation");
+  const blocksDetailTotalTrx = page.getByTestId("block.detail.overview.value.transactions");
+  const blocksDetailTrxFee = page.getByTestId("block.detail.overview.value.transactionFee");
+  const blocksDetailOutput = page.getByTestId("block.detail.overview.value.output");
+
+  //Block detail - Transactions Table
+  const blocksDetailTrxTableTxhash = page.getByTestId("block.detail.trxTable.txhash");
```
Comment (Luisdzc) [inline playwright/pages/blocks-dashboard.page.ts:41]: This locators should be in the block-detail.page.ts class

File: playwright/pages/blocks-dashboard.page.ts [hunk 3]
```diff
@@ -1,15 +1,252 @@
...
+      "Total fees in block widget to equal total fee in block on Blockfrost "
+    ).toEqual(+(blockFrostBlock?.fees || 0));
+
+    expect(
+      +((await blocksWidgetOutput.textContent())?.replaceAll(",", "") || 0) * 10 ** 6,
+      "Total output in block widget to equal total fee in block on Blockfrost "
+    ).toEqual(+(blockFrostBlock?.output || 0));
+  };
+
+  const checkTransactionsDetail = async () => {
```
Comment (Luisdzc) [inline playwright/pages/blocks-dashboard.page.ts:210]: When we check the redirect behavior as mention for the epochs tests we should also make a simple validation that the redirect goes to the respective element, in this case the user selects the transaction hash so when he be redirected we should also compare the selected tx hash is the same that is displayed in the transaction detail page and like mention for epochs test this check is executed over the transaction detail page so this function and respective locator should be in a separated page called transaction-detail.page.ts for example.

File: playwright/pages/blocks-dashboard.page.ts [hunk 4]
```diff
@@ -1,15 +1,252 @@
...
+    const url = await page.url();
+    await expect(deatailPageTitle, "Check title on transaction detail").toHaveText("Transaction Details");
+    expect(url, "Check url transaction detail").toContain("/transaction/");
+  };
+
+  const checkAddressDetail = async () => {
```
Comment (Luisdzc) [inline playwright/pages/blocks-dashboard.page.ts:216]: Same as above's comment

File: playwright/pages/blocks-dashboard.page.ts [hunk 5]
```diff
@@ -1,15 +1,252 @@
...
+    const url = await page.url();
+    await expect(addressDetailTitle, "Check title on address detail").toHaveText("Address Details");
+    expect(url, "Check url address detail").toContain("/address/");
+  };
+
+  const checkPoolDetail = async () => {
```
Comment (Luisdzc) [inline playwright/pages/blocks-dashboard.page.ts:222]: Same as above's comment

**Review by Luisdzc — CHANGES_REQUESTED** [review]

**Review by Luisdzc — APPROVED** [review]

**Review by obstar — APPROVED** [review]

**Review by cfuisat — APPROVED** [review]

**Pull request merged**